charts/super-agent/charts/super-agent-deployment/templates/_helpers.tpl

```diff
@@ -90,23 +170,130 @@ readOnlyRootFilesystem: true
 
 
 
-
-
 {{- /*
 Return .Values.config.auth.organizationId and fails if it does not exists
 */ -}}
 {{- define "newrelic-super-agent.auth.organizationId" -}}
-{{- if not ((.Values.config).auth).organizationId -}}
-  {{- fail ".Values.config.auth.organizationId is required." -}}
+{{- if ((.Values.config).auth).organizationId -}}
+  {{- .Values.config.auth.organizationId -}}
+{{- else -}}
+  {{- fail ".config.auth.organizationId is required." -}}
 {{- end -}}
-{{- .Values.config.auth.organizationId -}}
 {{- end -}}
 
 
 
 {{- /*
-Releases with "-auth" suffix.
+Check if .Values.config.auth.secret.name exists and use it to name auth' secret. If it does not exist, fallback to the name
+of the releases with "-auth" suffix.
 */ -}}
 {{- define "newrelic-super-agent.auth.secret.name" -}}
-  {{- include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "auth") }}
+{{- $secretName := ((((.Values.config).auth).secret).name) -}}
+{{- if $secretName -}}
+  {{- $secretName -}}
+{{- else -}}
+  {{- include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "auth" ) }}
+{{- end -}}
+{{- end -}}
+
+
+
+{{- /*
+Helper to toggle the creation of the job that creates and registers the system identity.
+*/ -}}
+{{- define "newrelic-super-agent.auth.secret.shouldRunJob" -}}
+{{- $privateKey := include "newrelic-super-agent.auth.secret.privateKey.data" . -}}
+{{- $clientId := include "newrelic-super-agent.auth.secret.clientId.data" . -}}
+
+{{- if and ((.Values.config).auth).enabled (((.Values.config).auth).secret).create (not $privateKey) (not $clientId) -}}
+  true
+{{- end -}}
+{{- end -}}
+
+
+
+{{- /*
+Helper to toggle the creation of the secret that has the system identity as values.
+*/ -}}
+{{- define "newrelic-super-agent.auth.secret.shouldTemplate" -}}
+{{- if and ((.Values.config).auth).enabled (((.Values.config).auth).secret).create -}}
+  {{- $privateKey := include "newrelic-super-agent.auth.secret.privateKey.data" . -}}
+  {{- $clientId := include "newrelic-super-agent.auth.secret.clientId.data" . -}}
+
+  {{- if and $privateKey $clientId -}}
+    true
+  {{- else if or $privateKey $clientId -}}
+    {{- fail "If you provide your own system identity data you have to provide both private key and client id" -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+
+
+{{- /*
+Check if .Values.config.auth.secret.private_key.secret_key exists and use it for the key in the secret containing the private
+key needed for the system identity. Fallbacks to `private_key`.
+*/ -}}
+{{- define "newrelic-super-agent.auth.secret.privateKey.key" -}}
+{{- $key := (((((.Values.config).auth).secret).private_key).secret_key) -}}
```

Comment:
`key` is confusing since it seems the `private_key`
also `secret_key` is not clear that is the key of the secret

MAybe a comment would help

Reply:
It is commented on the `values.yaml`: https://github.com/newrelic/helm-charts/blob/15207de405080f265cc3fec28af867a1f5b135e0/charts/super-agent/values.yaml#L141-L166

Do we need to comment it again?

Reply:
no, it is fine 😄